include/dali/imgcodec/image_decoder_interfaces.h

```diff
@@ -193,8 +195,24 @@ class DLL_PUBLIC ImageDecoderInstance {
                                              cspan<ROI> rois = {}) = 0;
   /**
    * @brief Sets a codec-specific parameter
+   *
+   * @return  true, if the parameter is relevant for this decoder, false otherwise
+   *
+   * @remarks The function may throw if the parameter name is valid for this decoder, but
+   *          the value is incorrect.
    */
   virtual bool SetParam(const char *key, const any &value) = 0;
+
+  /**
+   * @brief Sets codec-specific parameters.
+   *
+   * @return  The number of parameters that were relevant for this decoder.
+   *
+   * @remarks The function may throw if the parameter name is valid for this decoder, but
+   *          the value is incorrect.
```

Comment:
Should we add a remark here, that this function should be called in constructor if any of the SetParam/SetParams is overloaded in given decoder?

Reply:
It's not strictly necessary. What _should_ be added, though, is that this function should _not_ be called in a constructor of a base class.